gluon-activations/gluon-activations.md

```diff
@@ -0,0 +1,239 @@
+
+### Visualizing Activations
+In order to compare the various activation functions and to understand the nuances of their differences we have a snippet of code to to plot the activation functions and their gradients.
+
+
+```python
+import numpy as np
+import mxnet
+from mxnet import nd, gluon, autograd
+from matplotlib import pyplot as plt
+%matplotlib inline
+
+data = np.linspace(-10, 10, 501)
+
+def visualize_activation(data, activation_fn, activation):
+    x = nd.array(data)
+    x.attach_grad()
+    with autograd.record():
+        y = activation_fn(x)
+    y.backward()
+    
+    plt.figure()
+    plt.plot(x.asnumpy(), y.asnumpy())
+    plt.plot(x.asnumpy(), x.grad.asnumpy())
+    plt.legend(["{} activation".format(activation), "{} gradient".format(activation)])
+```
+
+## Sigmoids
+
+### Sigmoid
+The sigmoid activation function, also known as the logistic function or logit function, is perhaps the most widely known activation owing to its [long history](https://web.stanford.edu/class/psych209a/ReadingsByDate/02_06/PDPVolIChapter8.pdf) in neural network training and appearance in logistic regression and kernel methods for classification. 
+
+The sigmoid activation is a non-linear function that transforms any real valued input to a value between 0 and 1, giving it a natural probabilistic interpration. The sigmoid takes the form of the function below
+
+$$ \sigma(x) = \dfrac{1}{1 + e^x} $$ or alternatively
+
+$$ \sigma(x) = \dfrac{e^x}{e^x + 1} $$
+
+Note: the term sigmoid is overloaded and can be used to refer to the class of 's' shaped functions or particularly to the logistic function that we've just described. In MxNet the sigmoid activation specifically refers to logistic function sigmoid.
+
+
+
+```python
+visualize_activation(data, gluon.nn.Activation('sigmoid'), "Sigmoid")
+```
+
+
+![png](output_5_0.png)
+
+
+The sigmoid activation has since fallen out of use as the preferred activation function in designing neural networks due to some of it's properties, shown in the plot above, like not being zero-centered and inducing vanishing gradients, that leads to poor performance during neural network training. Vanishing gradients here refers to the tendency of the gradient of the sigmoid function to be nearly zero for most input values. 
+
+### tanh
+The tanh activation function is also an s shaped curve defined by the mathematical equation:
```

Comment:
Can you give an example of when to use before diving into the maths?

Reply:
okay. will add example